cpp/csp/python/PyEngine.h

```diff
@@ -55,6 +55,20 @@ class CSPIMPL_EXPORT PyEngine final: public PyObject
     Engine    * m_engine;
 };
 
+inline std::exception_ptr PyEngine_shutdownMakeException( PyObject * pyException )
+{
+    if( !PyExceptionInstance_Check( pyException ) )
+    {
+        PyObject* pyExceptionStr = PyObject_Str( pyException );
+        std::string pyExceptionString = PyUnicode_AsUTF8( pyExceptionStr );
+        std::string desc = "Expected Exception object as argument for shutdown_engine: got " + pyExceptionString + "of type " + Py_TYPE( pyException ) -> tp_name;
+        Py_DECREF(pyExceptionStr);
```

Comment:
spaces around brackets (but you should remove this anyways and use `PyObjectPtr` as that's easier)